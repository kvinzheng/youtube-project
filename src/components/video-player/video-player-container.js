import {connect} from 'react-redux';
import VideoPlayer from './video-player';
import {withRouter} from 'react-router-dom';

export const mapStateToProps = state => ({
  status: state.selectedVideo.status
    ? state.selectedVideo.status
    : 'PENDING',
  id: state.selectedVideo.video
    ? state.selectedVideo.video.id.videoId
    : '0',
  title: state.selectedVideo.video
    ? state.selectedVideo.video.snippet.title
    : 'title',
  channelTitle: state.selectedVideo.video
    ? state.selectedVideo.video.snippet.channelTitle
    : 'channelTitle',
  description: state.selectedVideo.video
    ? state.selectedVideo.video.snippet.description
    : 'description'
});

export default connect(mapStateToProps, null)(withRouter(VideoPlayer));
