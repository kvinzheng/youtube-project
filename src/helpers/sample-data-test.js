export const video = {
  "kind": "youtube#searchResult",
  "etag": "\"cbz3lIQ2N25AfwNr-BdxUVxJ_QY/Cdphs-sy_uELwbqUW-uWHjQcnqo\"",
  "id": {
    "kind": "youtube#video",
    "videoId": "hY7m5jjJ9mM"
  },
  "snippet": {
    "publishedAt": "2017-05-31T09:30:02.000Z",
    "channelId": "UC9obdDRxQkmn_4YpcBMTYLw",
    "title": "CATS will make you LAUGH YOUR HEAD OFF - Funny CAT compilation",
    "description": "HELP HOMELESS ANIMALS AND WIN A PRIZE BY CHOOSING AN AWESOME ITEM FROM OUR STORE: https://fanjoy.co/collections/tiger-productions ...",
    "thumbnails": {
      "default": {
        "url": "https://i.ytimg.com/vi/hY7m5jjJ9mM/default.jpg",
        "width": 120,
        "height": 90
      },
      "medium": {
        "url": "https://i.ytimg.com/vi/hY7m5jjJ9mM/mqdefault.jpg",
        "width": 320,
        "height": 180
      },
      "high": {
        "url": "https://i.ytimg.com/vi/hY7m5jjJ9mM/hqdefault.jpg",
        "width": 480,
        "height": 360
      }
    },
    "channelTitle": "Tiger FunnyWorks",
    "liveBroadcastContent": "none"
  }
};

export const videos = {
  "kind": "youtube#searchListResponse",
  "etag": "\"cbz3lIQ2N25AfwNr-BdxUVxJ_QY/qdDGutV8AE1VGqwVGO3M3L31giU\"",
  "nextPageToken": "CAUQAA",
  "regionCode": "US",
  "pageInfo": {
    "totalResults": 1000000,
    "resultsPerPage": 5
  },
  "items": [
    {
      "kind": "youtube#searchResult",
      "etag": "\"cbz3lIQ2N25AfwNr-BdxUVxJ_QY/Cdphs-sy_uELwbqUW-uWHjQcnqo\"",
      "id": {
        "kind": "youtube#video",
        "videoId": "hY7m5jjJ9mM"
      },
      "snippet": {
        "publishedAt": "2017-05-31T09:30:02.000Z",
        "channelId": "UC9obdDRxQkmn_4YpcBMTYLw",
        "title": "CATS will make you LAUGH YOUR HEAD OFF - Funny CAT compilation",
        "description": "HELP HOMELESS ANIMALS AND WIN A PRIZE BY CHOOSING AN AWESOME ITEM FROM OUR STORE: https://fanjoy.co/collections/tiger-productions ...",
        "thumbnails": {
          "default": {
            "url": "https://i.ytimg.com/vi/hY7m5jjJ9mM/default.jpg",
            "width": 120,
            "height": 90
          },
          "medium": {
            "url": "https://i.ytimg.com/vi/hY7m5jjJ9mM/mqdefault.jpg",
            "width": 320,
            "height": 180
          },
          "high": {
            "url": "https://i.ytimg.com/vi/hY7m5jjJ9mM/hqdefault.jpg",
            "width": 480,
            "height": 360
          }
        },
        "channelTitle": "Tiger FunnyWorks",
        "liveBroadcastContent": "none"
      }
    }, {
      "kind": "youtube#searchResult",
      "etag": "\"cbz3lIQ2N25AfwNr-BdxUVxJ_QY/u2n3wez7ljBkwPSV6WkGrkhsBlI\"",
      "id": {
        "kind": "youtube#video",
        "videoId": "5dsGWM5XGdg"
      },
      "snippet": {
        "publishedAt": "2016-12-24T10:30:00.000Z",
        "channelId": "UCKy3MG7_If9KlVuvw3rPMfw",
        "title": "Cats are so funny you will die laughing - Funny cat compilation",
        "description": "Cats are simply the funniest and most hilarious pets, they make us laugh all the time! Just look how all these cats & kittens play, fail, get along with dogs and ...",
        "thumbnails": {
          "default": {
            "url": "https://i.ytimg.com/vi/5dsGWM5XGdg/default.jpg",
            "width": 120,
            "height": 90
          },
          "medium": {
            "url": "https://i.ytimg.com/vi/5dsGWM5XGdg/mqdefault.jpg",
            "width": 320,
            "height": 180
          },
          "high": {
            "url": "https://i.ytimg.com/vi/5dsGWM5XGdg/hqdefault.jpg",
            "width": 480,
            "height": 360
          }
        },
        "channelTitle": "Tiger Productions",
        "liveBroadcastContent": "none"
      }
    }, {
      "kind": "youtube#searchResult",
      "etag": "\"cbz3lIQ2N25AfwNr-BdxUVxJ_QY/0ej2DuHdmsgJLqZkGmegnoBrISU\"",
      "id": {
        "kind": "youtube#video",
        "videoId": "tntOCGkgt98"
      },
      "snippet": {
        "publishedAt": "2013-12-31T05:21:57.000Z",
        "channelId": "UCMW0kSYo44uRxvE21JqEWTQ",
        "title": "Funny Cats Compilation [Most See] Funny Cat Videos Ever Part 1",
        "description": "Funny Cats Compilation - Funny Cat Videos - Funny Videos Check out the NEW Part 2 Compilation!: https://youtu.be/XHr2cvkA8rQ Funny Cats Compilation Try ...",
        "thumbnails": {
          "default": {
            "url": "https://i.ytimg.com/vi/tntOCGkgt98/default.jpg",
            "width": 120,
            "height": 90
          },
          "medium": {
            "url": "https://i.ytimg.com/vi/tntOCGkgt98/mqdefault.jpg",
            "width": 320,
            "height": 180
          },
          "high": {
            "url": "https://i.ytimg.com/vi/tntOCGkgt98/hqdefault.jpg",
            "width": 480,
            "height": 360
          }
        },
        "channelTitle": "Forget Your Sadness",
        "liveBroadcastContent": "none"
      }
    }, {
      "kind": "youtube#searchResult",
      "etag": "\"cbz3lIQ2N25AfwNr-BdxUVxJ_QY/z7OlabMtkYnYM6gpAKDiCkAEaEk\"",
      "id": {
        "kind": "youtube#video",
        "videoId": "njSyHmcEdkw"
      },
      "snippet": {
        "publishedAt": "2016-06-26T13:50:56.000Z",
        "channelId": "UCdmnOJ63TCe2AEdeFTYiweA",
        "title": "Funny Cats Compilation 2016  - Best Funny Cat Videos Ever || Funny Vines",
        "description": "CLICK NOW: https://goo.gl/TSNzGP for 25% discount T-Shirt in Sunfrog with code ILIKE25PLEEZ (Expired 31/10/17) 60$ OFF 200$ or more buyer total purchase ...",
        "thumbnails": {
          "default": {
            "url": "https://i.ytimg.com/vi/njSyHmcEdkw/default.jpg",
            "width": 120,
            "height": 90
          },
          "medium": {
            "url": "https://i.ytimg.com/vi/njSyHmcEdkw/mqdefault.jpg",
            "width": 320,
            "height": 180
          },
          "high": {
            "url": "https://i.ytimg.com/vi/njSyHmcEdkw/hqdefault.jpg",
            "width": 480,
            "height": 360
          }
        },
        "channelTitle": "Funny Vines",
        "liveBroadcastContent": "none"
      }
    }, {
      "kind": "youtube#searchResult",
      "etag": "\"cbz3lIQ2N25AfwNr-BdxUVxJ_QY/3BYpEDhVQCZ3B2FmPQVPbZK5FwI\"",
      "id": {
        "kind": "youtube#video",
        "videoId": "1Wh8RzcQZr4"
      },
      "snippet": {
        "publishedAt": "2017-05-02T12:00:00.000Z",
        "channelId": "UCPDis9pjXuqyI7RYLJ-TTSA",
        "title": "Hilarious Cat Fails (April 2017)",
        "description": "Enjoy these hilarious cat fails, some of them are just mean though!! Which feline is your favorite? Tell us in the comment section and don't forget to submit your ...",
        "thumbnails": {
          "default": {
            "url": "https://i.ytimg.com/vi/1Wh8RzcQZr4/default.jpg",
            "width": 120,
            "height": 90
          },
          "medium": {
            "url": "https://i.ytimg.com/vi/1Wh8RzcQZr4/mqdefault.jpg",
            "width": 320,
            "height": 180
          },
          "high": {
            "url": "https://i.ytimg.com/vi/1Wh8RzcQZr4/hqdefault.jpg",
            "width": 480,
            "height": 360
          }
        },
        "channelTitle": "FailArmy",
        "liveBroadcastContent": "none"
      }
    }
  ]
};

export const sampleData = {
  search: {
    picture: 'dog'
  },
  selectedVideo: {
    video: video,
    status: 'FULFILLED'
  },
  youtubeData: {
    myData: videos,
    status: 'FULFILLED'
  }
}
export const emptySampleData = {
  search: {
    picture: null,
  },
  selectedVideo: {
    video: null,
    status: null,
  },
  youtubeData: {
    myData: null,
    status: null,
  }
}
