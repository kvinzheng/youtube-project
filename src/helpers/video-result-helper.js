export const dataConvertor = array =>
  array.map(ele => ele.snippet);
