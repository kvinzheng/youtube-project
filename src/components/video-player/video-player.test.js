import React from 'react';
import {shallow, configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import VideoPlayer from './video-player';
import {mapStateToProps} from './video-player-container';
import {sampleData} from '../../helpers/sample-data-test';

configure({adapter: new Adapter()});

const history = {
  push: jest.fn()
}

describe('SearchBar Component', () => {
  it('should render without any data defined', () => {
    const component = shallow(<VideoPlayer status={''} id={''} title={''} channelTitle={''} description={''}/>);
    expect(component).toMatchSnapshot();
  });

  it('should render with props passed in', () => {
    const component = shallow(<VideoPlayer status={'FULFILLED'} id={'2'} title={'cat is cool'} channelTitle={'pet shop'} description={'cat is very cool'}/>);
    expect(component).toMatchSnapshot();
  });
  it('Renders with correct classes', () => {
    const component = shallow(<VideoPlayer status={'FULFILLED'} id={'2'} title={'cat is cool'} channelTitle={'pet shop'} description={'cat is very cool'}/>);
    expect(component.find('.video-player').length).toBe(1);
    expect(component.find('.player').length).toBe(1);
    expect(component.find('.back-to-home').length).toBe(1);
    expect(component.find('.videoTitle').length).toBe(1);
    expect(component.find('.videoDescription').length).toBe(1);
  });

  it('mapStateToProps', () => {
    const state = sampleData;
    const expected = {
      status: 'FULFILLED',
      id: 'hY7m5jjJ9mM',
      description: 'HELP HOMELESS ANIMALS AND WIN A PRIZE BY CHOOSING AN AWESOME ITEM FROM OUR STORE: https://fanjoy.co/collections/tiger-productions ...',
      title: 'CATS will make you LAUGH YOUR HEAD OFF - Funny CAT compilation',
      channelTitle: 'Tiger FunnyWorks'
    };
    expect(mapStateToProps(state)).toEqual(expected);
  });

  it('mapStateToProps default data', () => {
    const state = {
      search: {
        picture: null
      },
      selectedVideo: {
        video: null,
        state: 'PENDING'
      },
      youtubeData: {
        myData: null,
        status: null
      }
    };
    const expected = {
      status: 'PENDING',
      id: '0',
      description: 'description',
      title: 'title',
      channelTitle: 'channelTitle'
    };
    expect(mapStateToProps(state)).toEqual(expected);
  });

  it('When you click on the button', () => {
    const component = mount(<VideoPlayer status={'FULFILLED'} id={'2'} title={'cat is cool'} channelTitle={'pet shop'} description={'cat is very cool'} history={history}/>);
    component.find('.back-to-home').simulate('click', {
      preventDefault: () => {}
    });
    expect(history.push).toHaveBeenCalled();
    expect(history.push.mock.calls[0][0]).toBe('/');
  });
});
