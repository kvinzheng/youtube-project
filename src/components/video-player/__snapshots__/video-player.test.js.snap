// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component should render with props passed in 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VideoPlayer
    channelTitle="pet shop"
    description="cat is very cool"
    id="2"
    status="FULFILLED"
    title="cat is cool"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="video-player"
>
          <div>
                    <iframe
                              className="player"
                              src="https://www.youtube.com/embed/2"
                              title="cat is cool"
                    />
          </div>
          <div
                    className="video-description"
          >
                    <h3
                              className="video-title"
                    >
                              cat is cool
                    </h3>
                    <h4>
                              Channel: pet shop
                    </h4>
                    <p>
                              cat is very cool
                    </p>
          </div>
</div>,
        <button
          className="back-to-home"
          onClick={[Function]}
>
          Back To Youtube Home Page
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <iframe
                            className="player"
                            src="https://www.youtube.com/embed/2"
                            title="cat is cool"
              />
</div>,
            <div
              className="video-description"
>
              <h3
                            className="video-title"
              >
                            cat is cool
              </h3>
              <h4>
                            Channel: pet shop
              </h4>
              <p>
                            cat is very cool
              </p>
</div>,
          ],
          "className": "video-player",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <iframe
                className="player"
                src="https://www.youtube.com/embed/2"
                title="cat is cool"
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "className": "player",
                "src": "https://www.youtube.com/embed/2",
                "title": "cat is cool",
              },
              "ref": null,
              "rendered": null,
              "type": "iframe",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3
                  className="video-title"
>
                  cat is cool
</h3>,
                <h4>
                  Channel: pet shop
</h4>,
                <p>
                  cat is very cool
</p>,
              ],
              "className": "video-description",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "cat is cool",
                  "className": "video-title",
                },
                "ref": null,
                "rendered": "cat is cool",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Channel: pet shop",
                },
                "ref": null,
                "rendered": "Channel: pet shop",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "cat is very cool",
                },
                "ref": null,
                "rendered": "cat is very cool",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Back To Youtube Home Page",
          "className": "back-to-home",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Back To Youtube Home Page",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="video-player"
>
            <div>
                        <iframe
                                    className="player"
                                    src="https://www.youtube.com/embed/2"
                                    title="cat is cool"
                        />
            </div>
            <div
                        className="video-description"
            >
                        <h3
                                    className="video-title"
                        >
                                    cat is cool
                        </h3>
                        <h4>
                                    Channel: pet shop
                        </h4>
                        <p>
                                    cat is very cool
                        </p>
            </div>
</div>,
          <button
            className="back-to-home"
            onClick={[Function]}
>
            Back To Youtube Home Page
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <iframe
                                className="player"
                                src="https://www.youtube.com/embed/2"
                                title="cat is cool"
                />
</div>,
              <div
                className="video-description"
>
                <h3
                                className="video-title"
                >
                                cat is cool
                </h3>
                <h4>
                                Channel: pet shop
                </h4>
                <p>
                                cat is very cool
                </p>
</div>,
            ],
            "className": "video-player",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <iframe
                  className="player"
                  src="https://www.youtube.com/embed/2"
                  title="cat is cool"
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "className": "player",
                  "src": "https://www.youtube.com/embed/2",
                  "title": "cat is cool",
                },
                "ref": null,
                "rendered": null,
                "type": "iframe",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3
                    className="video-title"
>
                    cat is cool
</h3>,
                  <h4>
                    Channel: pet shop
</h4>,
                  <p>
                    cat is very cool
</p>,
                ],
                "className": "video-description",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "cat is cool",
                    "className": "video-title",
                  },
                  "ref": null,
                  "rendered": "cat is cool",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Channel: pet shop",
                  },
                  "ref": null,
                  "rendered": "Channel: pet shop",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "cat is very cool",
                  },
                  "ref": null,
                  "rendered": "cat is very cool",
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Back To Youtube Home Page",
            "className": "back-to-home",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Back To Youtube Home Page",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SearchBar Component should render without any data defined 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VideoPlayer
    channelTitle=""
    description=""
    id=""
    status=""
    title=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="back-to-home-container"
>
           Please go back to home page and select a video
</div>,
        <button
          className="back-to-home"
          onClick={[Function]}
>
          Back To Youtube Home Page
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": " Please go back to home page and select a video",
          "className": "back-to-home-container",
        },
        "ref": null,
        "rendered": " Please go back to home page and select a video",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Back To Youtube Home Page",
          "className": "back-to-home",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Back To Youtube Home Page",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="back-to-home-container"
>
             Please go back to home page and select a video
</div>,
          <button
            className="back-to-home"
            onClick={[Function]}
>
            Back To Youtube Home Page
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": " Please go back to home page and select a video",
            "className": "back-to-home-container",
          },
          "ref": null,
          "rendered": " Please go back to home page and select a video",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Back To Youtube Home Page",
            "className": "back-to-home",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Back To Youtube Home Page",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
